compiler/rustc_hir_analysis/src/hir_ty_lowering/mod.rs

```diff
@@ -1020,99 +1022,106 @@ impl<'tcx> dyn HirTyLowerer<'tcx> + '_ {
                 if let Some(variant_def) = variant_def {
                     if permit_variants {
                         tcx.check_stability(variant_def.def_id, Some(hir_ref_id), span, None);
-                        self.prohibit_generic_args(slice::from_ref(assoc_segment).iter(), |err| {
-                            err.note("enum variants can't have type parameters");
-                            let type_name = tcx.item_name(adt_def.did());
-                            let msg = format!(
-                                "you might have meant to specify type parameters on enum \
+                        let _ = self.prohibit_generic_args(
+                            slice::from_ref(assoc_segment).iter(),
+                            |err| {
```

Comment:
can you also extract this closure into a function. It's quite long and already at a pretty high identation

Reply:
> can you also extract this closure into a function. It's quite long and already at a pretty high identation

Done. Thank you.